### DISCUSSION TITLE
How to get Code/Text version of Auto Index

### DISCUSSION THREAD

**Discussion by mfonnesbeck** [ravendb/ravendb#17056] [Q&A]
I created a query in the cloud-based Raven Studio that I was developing and it created an auto index for that query. I wanted to now turn that auto index into a named static index that I could query from my code. However, when I view the auto index it shows it as a sort of flow chart. Is there a way to see or turn this auto index into a code version that I could create in the studio as a static index and then also in my C# code to query against?

**Comment by ayende** [comment]
Hi, there is no way to do that through the studio.
Typically you would just write this index directly.

Auto indexes are basically:

```
from doc in docs.Collection
select new
{
   doc.Name,
  // other fields go here
}```

**Comment by mfonnesbeck** [comment]
Ok. Thanks.

This auto index is a little more complex. It contains a group by, where, order by, and then select clauses.  So it shows the collection, a group by, and then the result in the flow chart.

**Reply by ayende** [reply]
In that case, this is a map/reduce index. See the docs for details

**Reply by mfonnesbeck** [reply]
Ok. I got it to work as an actual index. However, I am not getting results I am desiring. 

I produce the map using a Community Id, Closed Timestamp, and a count value of 1 for each document. Then I reduce the map results and group on the Community Id and Closed Timestamp, however, I am wanting a summed count of documents by days and the closed timestamp is a DateTime. I cannot seem to just pull out the Date part of the Closed Timestamp and group on individual whole days and cannot find help in the docs to this situation.

Recommendations?

Map: 
from ticket in docs.tickets
     where ticket.ClosedTimestamp != DateTime.MinValue
     select new {
         CommunityId = ticket.CommunityId,
         ClosedTimestamp = ticket.ClosedTimestamp,
         Count = 1
     }

Reduce:
from result in results
    group result by 
        new { result.CommunityId, result.ClosedTimestamp } into t
    select new
    {
        CommunityId = t.Key.CommunityId,
        ClosedTimestamp = t.Key.ClosedTimestamp,
        Count = t.Sum(x => x.Count)
    }

**Reply by ayende** [reply]
`ClosedTimestamp = ticket.ClosedTimestamp.Date,` in the map should do that, no?

**Reply by mfonnesbeck** [reply]
I was able to add it to the Reduce, but not the Map. It kept giving me a compiler error stating that .Date property or method did not exist.

**Reply by ayende** [reply]
It has to be on the map, not the reduce.
Utterly unable to help without a full repro
